solidity/contracts/proxy/DepositFactory.sol

```diff
@@ -19,9 +21,9 @@ contract DepositFactory is CloneFactory, TBTCSystemAuthority{
     // which will be used as a master contract for cloning.
     address payable public masterDepositAddress;
     TBTCDepositToken tbtcDepositToken;
-    address public tbtcSystem;
-    address public tbtcToken;
-    address public feeRebateToken;
+    TBTCSystem public tbtcSystem;
+    TBTCToken public tbtcToken;
+    FeeRebateToken public feeRebateToken;
     address public vendingMachine;
```

Comment:
Should we hit this one?

Reply:
We only use the address and never call any `VendingMachine` contract functions.
Hitting this would be counterproductive since we would just have to cast it back to use it.

Reply:
Why do we use the address? `createNewDeposit` should likewise take a `VendingMachine`, right? Generally eschewing the type check is a bad idea, and if we have to cast to a looser type at the use point, that's fine---otherwise you're leaving on the table the capacity of the compiler to bonk you when you mishandle some point in the initialization chain.

Reply:
> Why do we use the address? 

It's just used for redemption logic:
` if(tdtHolder == vendingMachine){...}`

Worth noting, It's introduced into the system as a contract type via tbtcSystem.initialize().
Then it's passed along and stored as an address. Might be worth renaming it to `vendingMachineAddress` to avoid confusion?

Reply:
I like this. Makes it clear that we're really only interested in the address.

Also makes it clear this is a dirty dirty hack and in another life where we had more time we'd try to get around it haha.

Reply:
ok awesome